game/src/graphics/darkness.rs

```diff
@@ -0,0 +1,395 @@
+                file!(),
+                line!()
+            );
+            return;
+        };
+
+        render_app
+            // Bevy's renderer uses a render graph which is a collection of nodes in a directed acyclic graph.
+            // It currently runs on each view/camera and executes each node in the specified order.
+            // It will make sure that any node that needs a dependency from another node
+            // only runs when that dependency is done.
+            //
+            // Each node can execute arbitrary work, but it generally runs at least one render pass.
+            // A node only has access to the render world, so if you need data from the main world
+            // you need to extract it manually or with the plugin like above.
+            // Add a [`Node`] to the [`RenderGraph`]
+            // The Node needs to impl FromWorld
+            //
+            // The [`ViewNodeRunner`] is a special [`Node`] that will automatically run the node for each view
+            // matching the [`ViewQuery`]
+            .add_render_graph_node::<ViewNodeRunner<DarknessPostProcessNode>>(
+                // Specify the name of the graph, in this case we want the graph for 3d
+                core_2d::graph::NAME,
+                // It also needs the name of the node
+                DarknessPostProcessNode::NAME,
+            )
+            .add_render_graph_edges(
+                core_2d::graph::NAME,
+                // Specify the node ordering.
+                // This will automatically create all required node edges to enforce the given ordering.
+                // Currently runs after ToneMapping, which seems to give best appearance... might need to revisit
+                // to handle bloom/ other glowing objects.
+                &[
+                    core_2d::graph::node::TONEMAPPING,
+                    DarknessPostProcessNode::NAME,
+                    core_2d::graph::node::END_MAIN_PASS_POST_PROCESSING,
+                ],
+            );
+    }
+
+    fn finish(&self, app: &mut App) {
+        // We need to get the render app from the main app
+        let Ok(render_app) = app.get_sub_app_mut(RenderApp) else {
+            return;
+        };
+
+        render_app
+            // Initialize the pipeline
+            .init_resource::<DarknessPostProcessPipeline>();
+    }
+}
+
+#[derive(Component, Default, Clone, Copy, ExtractComponent, ShaderType)]
+/// Darkness Post Process Settings
+pub struct DarknessSettings {
+    /// 0.0 is pitch black, and 1.0 is normal brightness
+    pub bg_light_level: f32,
+    /// Position is currently relative to the camera center
+    pub lantern_position: Vec2,
+    /// Output of the light source; 0.0 is off and 1.0 is normal brightness.
+    pub lantern: f32,
+    /// RGB
+    pub lantern_color: Vec3,
+    /// RGB
+    pub bg_light_color: Vec3,
+    // WebGL2 structs must be 16 byte aligned.
+    #[cfg(feature = "webgl2")]
+    _webgl2_padding: Vec2,
```

Comment:
We don't really need to worry about webgl2 compat for this game, but sure why not.

Reply:
Was part of the post process example; figured may as well leave it in